Requester Pays Best Practice (#1021)
radiantearth/stac-spec: best-practices.md

design to look and feel like your main web presence. STAC aims to be decentralized, so each STAC-compliant data catalog 
should have its own location and just be part of the wider web.

### Requester Pays

It is very common that large, freely available datasets are set up with a 'requester pays' configuration. This is an option 

> [on AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/RequesterPaysBuckets.html)) and [on 

Reviewer: ```suggestion
[on AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/RequesterPaysBuckets.html) and [on 
```

> Google Cloud](https://cloud.google.com/storage/docs/requester-pays), that enables data providers to make their data freely

Reviewer 2: ```suggestion
Google Cloud](https://cloud.google.com/storage/docs/requester-pays), that enables data providers to make their data
```

> available to everyone, while placing the cost of '[egress](https://www.hostdime.com/blog/data-egress-fees-cloud/)' 

Reviewer 2: ```suggestion
available to everyone, while the cloud platform charges access costs (such as per-request and data '[egress](https://www.hostdime.com/blog/data-egress-fees-cloud/)') to the user accessing the data.
```

> on to the user. For popular datasets that are large in size the egress costs can be substantial, to the point where much

Reviewer 2: ```suggestion
For popular datasets that are large in size the egress costs can be substantial, to the point where much
```

less data would be available if the cost of distribution was always on the data provider.

> For data providers using STAC with requester pays buckets there are two main recommendations:

Reviewer 2: ```suggestion
For data providers using STAC with requester pays buckets, there are two main recommendations:
```

> 1) Put the STAC JSON in a separate bucket that is public for everyone, not requestor pays. This enables the STAC metadata

Reviewer 2: ```suggestion
1) Put the STAC JSON in a separate bucket that is public for everyone and **not** requestor pays. This enables the STAC metadata
```

to be far more crawlable and searchable, but the cost of the egress of STAC files should be miniscule compared to that of

> the actual data. The STAC community can help you work with cloud providers for potential free hosting if you are doing open

Reviewer 2: I'm a little hesitant to put this in here, since AWS wouldn't support the MODIS opendata bucket.

Author: I wasn't aware of that. I'm happy to help push on that one with AWS. And I think that between Microsoft and Google it's likely to get some free hosting. I think it may be worth leaving in so we can at least try as a community, but I don't think it's on a specific provider. I think I also meant it to be more on the 'stac only' free hosting.

Reviewer 2: Okay, I'm fine with as long as it's something that can actually happen.

AWS (Flasher) passed on it b/c of some other work NASA/USGS was doing to make data available on AWS, which I don't ever actually remember seeing realized.

Author: Ok, I just updated the language a bit - the other key point is we can try with microsoft, google and aws. I think at this moment microsoft is the 'best', and I assume that will shift between the three over time, and perhaps there will be others as well.

data as requestor pays and aren't able to pay the costs of a completely open STAC bucket, as they are most all supportive of
STAC.

> 2) Don't use `http://` style urls if the provider has a specific protocol (specifically `s3://` on AWS and `gs://` on Google

Reviewer 2: ```suggestion
2) For Asset href values to resources in a requestor pays bucket, use the cloud provider-specific protocol (e.g., `s3://` on AWS and `gs://` on Google Cloud) instead of an `https://` url. 
```

> Cloud). Most every client will just fail on an `http://` link that is requestor pays, but most clients that understand

Reviewer 2: ```suggestion
Cloud). Most clients do not have special handling for `https://` links to cloud provider resources that require a requestor pays flag and authentication, so they simply fail. Many clients do have special handling for `s3://` or `gs://` URLs that will add a requestor pays parameter and apply appropriate authentication to the request.
```

> 2) For Asset href values to resources in a requestor pays bucket, use the cloud provider-specific protocol (e.g., `s3://` on AWS and `gs://` on Google Cloud) instead of an `https://` url. 
> Cloud). Most clients do not have special handling for `https://` links to cloud provider resources that require a requestor pays flag and authentication, so they simply fail. Many clients do have special handling for `s3://` or `gs://` URLs that will add a requestor pays parameter and apply appropriate authentication to the request.

Reviewer: ```suggestion
2) For Asset href values to resources in a requestor pays bucket, use the cloud provider-specific protocol (e.g., `s3://` on AWS and `gs://` on Google Cloud) instead of an `https://` url. Most clients do not have special handling for `https://` links to cloud provider resources that require a requestor pays flag and authentication, so they simply fail. Many clients have special handling for `s3://` or `gs://` URLs that will add a requestor pays parameter and will apply appropriate authentication to the request.
```

> the cloud-specific protocols will at least have an option to register a paid account and properly charge for access. 

Reviewer: ```suggestion
Using cloud-specific protocols will at least give users an option to register a paid account and allow the data provider to properly charge for access. 
```

> STAC-specific tools in turn can look for the cloud-specific protocols and know to use requestor pays. 

Reviewer: ```suggestion
STAC-specific tools in turn can look for the cloud-specific protocols and know to use the requestor pays feature for that specific cloud platform.
```
